Modix.Services/Core/RolesLogMessages.cs

```diff
@@ -0,0 +1,135 @@
+﻿#nullable enable
+
+using System;
+
+using Microsoft.Extensions.Logging;
+
+namespace Modix.Services.Core
+{
+    internal static class RolesLogMessages
+    {
+        public enum EventType
+        {
+            RoleTracking            = 0x0001,
+            RoleTracked             = 0x0002,
+            RoleUpdating            = 0x0003,
+            RoleUpdated             = 0x0004,
+            RoleUpdateFailed        = 0x0005,
+            RoleCreating            = 0x0006,
+            RoleCreated             = 0x0007,
+            TransactionBeginning    = 0x0008,
+            TransactionComitting    = 0x0009
+        }
+
+        public static IDisposable BeginRoleScope(
+                ILogger logger,
+                ulong guildId,
+                ulong roleId)
+            => _beginRoleScope.Invoke(
+                logger,
+                guildId,
+                roleId);
+        private static readonly Func<ILogger, ulong, ulong, IDisposable> _beginRoleScope
+            = LoggerMessage.DefineScope<ulong, ulong>(
```

Comment:
I'm not seeing this actually write out to the console. Seems like we need a little extra configuration to make that happen: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0#log-scopes

Reply:
Yeah, scopes are ignored by all of our currently-used logging sinks. Mainly because none of them are structured. An overhaul of our logging sinks is definitely on my soon-to-do list.

Reply:
That's fair. In the meanwhile, these logs won't feel quite as helpful, since the scope ones are the only ones that actually tell you the IDs